### ISSUE TITLE
genome with >0.99 ANI and 1 aligned fraction in different secondary clusters

### ISSUE THREAD

**Issue by chassenr** [MrOlm/drep#192]
Hi @MrOlm ,

I am using dRep to get a non-redundant set of bins from the same data set, which was assembled and binned with different approaches. I am using dRep version 3.2.2 (snakemake had problems installing the newer versions) with the following command:

```
dRep dereplicate -g {output.binlist} --genomeInfo {output.genomeInfo} --length 50000 -comp 50 -con 10 --S_algorithm fastANI -pa 0.85 -sa 0.95 -nc 0.5 -p {threads} {params.outdir}
```

I noticed that genomes, that according to the Ndb.csv had an ANI of 0.996207 with an aligned fraction (AF) of 1.0, were placed into different secondary clusters. In this example, there were 5 genomes, which according to ANI and AF alone should have been in the same cluster, but were separate into separate secondary clusters, so this behavior of dRep was not an exception.

Do you have an idea what could be happening here? 

Thank you very much!

Kind regards,
Christiane

**Comment by MrOlm** [comment]
Hi Christiane,

Thanks for reaching out. It sounds like these genomes were in the same primary cluster, so it probably has something to do with the oddities of hierarchical clustering (https://drep.readthedocs.io/en/latest/choosing_parameters.html#oddities-of-hierarchical-clustering). My guess is that having `-nc 0.5` is at the heart of the problem; see https://drep.readthedocs.io/en/latest/choosing_parameters.html#minimum-alignment-coverage for a little bit of my thoughts on the issue. The `-nc` argument is just a difficult thing to work with with average hierarchal clustering; these 5 genomes probably have > 0.5 AF to some genomes and < 0.5 AF to others. If I were you I would just leave the -nc at the default value and my guess is this issue will go away.

Best,
Matt

**Comment by chassenr** [comment]
Hi Matt,

thanks for your super fast reply (and sorry for taking a bit longer to get back to you). I have tested different ```-nc``` settings,  also revisiting some old data sets of mine. Using the default, i.e. ```-S fastANI -nc 0.1``` improved the situation somewhat, but I was still not convinced by the results, when comparing them to an independent fastANI run, manually checking pairwise similarity in the output. Since in the past I ran dRep with ```-S ANImf``` I also compared the results at the same ```-nc``` between the two different algorithms. I was not expecting such a large difference in performance: ```-S ANImf -nc 0.5``` worked much better and produced more realistic clusters than ```-S fastANI -nc 0.1```. What made you decide to set the default algorithm to fastANI in the latest versions of dRep? Is it possible that the difference in behavior between fastANI and ANImf that I observed are version related, although based on the information in the changelog, I did not find an indication for this?

Thanks!

Best,
Christiane

**Comment by MrOlm** [comment]
Hi Christiane,

Thanks for looking into this so closely and getting back to me. I have not heard many reports of ANImf outperforming fastANI, so this is very valuable information.

I made fastANI the default in an update that was aimed and allowing dRep to be used with massive datasets, since fastANI is a good bit faster than ANImf. I assumed there would be some tradeoff in accuracy, but I did not know it was substantial like this. 

I really doubt any changes in dRep would impact this at all- it's been a very long time since I've changed any of the core clustering code in dRep.

Best,
Matt

**Comment by chassenr** [comment]
Hi Matt, 

just FYI, at ```-nc 0.5``` and otherwise identical settings fastANI produced 1218 secondary clusters, whereas ANImf only returned 507. I was only working with a few thousands (and not tens of thousands) of genomes, so ANImf was ok for me. I do understand the switch to fastANI, though. 

I also ran fastANI outside of dRep to confirm the dRep clustering with an independent (network-based) clustering approach. With this, the fastANI-generated values for ANI and aligned fraction resulted in a clustering that was very comparable to that of dRep with ANImf and ```-nc 0.5``` (which upon manual inspection outperformed by improvised clustering). I therefore assume that the reason for the diverging dRep output may not be the fastANI run itself...

If you want to pursue this further, I can share the data_tables of the dRep output and the scripts of my tests (and if required also the input genomes, although those only confidentially). The data set that I worked with included some low completeness genomes, so I guess this might also have an impact, but I would still have expected more similar results...

Cheers,
Christiane
